comp/otelcol/provider/component.go

```diff
@@ -0,0 +1,10 @@
+package provider
+
+import "github.com/DataDog/datadog-agent/comp/otelcol/provider/providerimpl"
+
+// team: opentelemetry
+
+// Component is the component type.
+type Component interface {
+	providerimpl.ExtendedConfigProvider
```

Comment:
Can we define ExtendedConfigProvider in this file?

Can we follow new component structure ? use 
`provider/def` -> for component and other type definitions 
`provider/fx` -> to define module
`provider/impl` -> for implementation . 

Please see contribFx component or collector component

Reply:
Followed new component structure and Moved `ExtendedConfigProvider` definition to `provider/def`.